README.md

```diff
@@ -1,51 +1,199 @@
+        let voipRegistry: PKPushRegistry = PKPushRegistry(queue: mainQueue)
+        voipRegistry.delegate = self
+        voipRegistry.desiredPushTypes = [PKPushType.voIP]
+        
+        return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+    }
+    
+    // Call back from Recent history
+    override func application(_ application: UIApplication,
+                              continue userActivity: NSUserActivity,
+                              restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
+        
+        guard let handleObj = userActivity.handle else {
+            return false
+        }
+        
+        guard let isVideo = userActivity.isVideo else {
+            return false
+        }
+        let callerName = handleObj.getDecryptHandle()["callerName"] as? String ?? ""
+        let handle = handleObj.getDecryptHandle()["handle"] as? String ?? ""
+        let data = flutter_callkeep.Data(id: UUID().uuidString, callerName: callerName, handle: handle, hasVideo: isVideo)
+        //set more data...
+        data.callerName = "Johnny"
+        SwiftCallKeepPlugin.sharedInstance?.startCall(data, fromPushKit: true)
+        
+        return super.application(application, continue: userActivity, restorationHandler: restorationHandler)
+    }
+    
+    // Handle updated push credentials
+    func pushRegistry(_ registry: PKPushRegistry, didUpdate credentials: PKPushCredentials, for type: PKPushType) {
+        print(credentials.token)
+        let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
+        print(deviceToken)
+        //Save deviceToken to your server
+        SwiftCallKeepPlugin.sharedInstance?.setDevicePushTokenVoIP(deviceToken)
+    }
+    
+    func pushRegistry(_ registry: PKPushRegistry, didInvalidatePushTokenFor type: PKPushType) {
+        print("didInvalidatePushTokenFor")
+        SwiftCallKeepPlugin.sharedInstance?.setDevicePushTokenVoIP("")
+    }
+    
+    // Handle incoming pushes
+    func pushRegistry(_ registry: PKPushRegistry, didReceiveIncomingPushWith payload: PKPushPayload, for type: PKPushType, completion: @escaping () -> Void) {
+        print("didReceiveIncomingPushWith")
+        guard type == .voIP else { return }
+        
+        let id = payload.dictionaryPayload["id"] as? String ?? ""
+        let callerName = payload.dictionaryPayload["callerName"] as? String ?? ""
+        let handle = payload.dictionaryPayload["handle"] as? String ?? ""
+        let isVideo = payload.dictionaryPayload["isVideo"] as? Bool ?? false
+        
+        let data = flutter_callkeep.Data(id: id, callerName: callerName, handle: handle, hasVideo: isVideo)
+        //set more data
+        data.extra = ["user": "abc@123", "platform": "ios"]
+        //data.iconName = ...
+        //data.appName = ...
+        //data.....
+        SwiftCallKeepPlugin.sharedInstance?.displayIncomingCall(data, fromPushKit: true)
+    }   
+}
```

Comment:
This part seems to hard to read i think better to adding inner changes for each function

Reply:
I agree, it also a lot of code you have to add manually, we can improve on it after this PR.